src/radical/pilot/agent/radical-pilot-agent-multicore.py

```diff
@@ -5429,60 +5445,114 @@ def initialize(self):
 
     # --------------------------------------------------------------------------
     #
-    def finalize(self):
+    def alive_barrier(self):
+
+        # FIXME: wait for bridges, too?  But we need pubsub for counting... Duh!
+        total = len(self._bridges)    + \
+                len(self._components) + \
+                len(self._workers   ) + \
+                len(self._sub_agents)
+        start   = time.time()
+        timeout = 60
```

Comment:
Hmm, probably not a good value on BW?  Any suggestion?

The timeout should actually not strictly be necessary anymore, as we watch the process handles now  and will detect if any component fails before it can send an alive message.  That does not hold for the bridges though (we need the channels to send alive in the first place... :P).  So, we could reduce the timeout scope to the bridges -- which makes this code somewhat more convoluted though...